server/plugin/internal/graphql/client.go

```diff
@@ -0,0 +1,60 @@
+package graphql
+
+import (
+	"context"
+	"net/url"
+	"path"
+
+	"github.com/mattermost/mattermost-server/v6/plugin"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+// Client encapsulates the third party package that communicates with Github GraphQL API
+type Client struct {
+	client   *githubv4.Client
+	org      string
+	username string
+	api      plugin.API
+}
+
+// NewClient creates and returns Client. Third party package that queries GraphQL is initialized here.
+func NewClient(api plugin.API, token oauth2.Token, username, orgName, enterpriseBaseURL string) *Client {
+	ts := oauth2.StaticTokenSource(&token)
+	httpClient := oauth2.NewClient(context.Background(), ts)
+	var client Client
+
+	if enterpriseBaseURL == "" || orgName == "" {
+		client = Client{
+			username: username,
+			client:   githubv4.NewClient(httpClient),
+			api:      api,
+		}
+	} else {
+		baseURL, err := url.Parse(enterpriseBaseURL)
+		if err != nil {
+			api.LogDebug("Not able to parse the URL", "Error", err.Error())
+		}
+
+		baseURL.Path = path.Join(baseURL.Path, "api", "graphql")
```

Comment:
This line can have nil pointer dereference error as we are not returning in the above `if err != nil` condition. In that case, `baseURL` can be nil.